src/instruments/src/EFB/Ground/Pages/ServicesPage.tsx

```diff
@@ -38,159 +43,524 @@ const ServiceButtonWrapper: FC<ServiceButtonWrapperProps> = ({ children, classNa
+                }
+            }, 5000);
         }
     };
 
+    // Doors
+    useEffect(() => {
+        simpleServiceListenerHandling(cabinDoorButtonState, setCabinDoorButtonState, cabinDoorOpen);
+        simpleServiceListenerHandling(cargoDoorButtonState, setCargoDoorButtonState, cargoDoorOpen);
+        simpleServiceListenerHandling(aftDoorButtonState, setAftDoorButtonState, aftDoorOpen);
+    }, [cabinDoorOpen, cargoDoorOpen, aftDoorOpen]);
+
+    // Fuel
+    useEffect(() => {
+        simpleServiceListenerHandling(fuelTruckButtonState, setFuelTruckButtonState, fuelingActive);
+    }, [fuelingActive]);
+
+    // Gpu
+    useEffect(() => {
+        simpleServiceListenerHandling(gpuButtonState, setGpuButtonState, gpuActive);
+    }, [gpuActive]);
+
+    // Cabin Door listener for JetBridge Button
+    useEffect(() => {
+        complexServiceListenerHandling(
+            jetWayButtonStateRef,
+            setJetWayButtonState,
+            cabinDoorButtonState,
+            setCabinDoorButtonState,
+            cabinDoorOpen,
+        );
+    }, [cabinDoorOpen]);
+
+    // Cargo Door listener for Baggage Button
+    useEffect(() => {
+        complexServiceListenerHandling(
+            baggageButtonStateRef,
+            setBaggageButtonState,
+            cargoDoorButtonState,
+            setCargoDoorButtonState,
+            cargoDoorOpen,
+        );
+    }, [cargoDoorOpen]);
+
+    // Aft Cabin Door listener fo rCatering Button
+    useEffect(() => {
+        complexServiceListenerHandling(
+            cateringButtonStateRef,
+            setCateringButtonState,
+            aftDoorButtonState,
+            setAftDoorButtonState,
+            aftDoorOpen,
+        );
+    }, [aftDoorOpen]);
+
+    // Pushback or movement start --> disable buttons and close doors
+    // Enable buttons if all have been disabled before
+    useEffect(() => {
+        if (!groundServicesAvailable) {
+            dispatch(setCabinDoorButtonState(ServiceButtonState.DISABLED));
+            dispatch(setJetWayButtonState(ServiceButtonState.DISABLED));
+            dispatch(setFuelTruckButtonState(ServiceButtonState.DISABLED));
+            dispatch(setGpuButtonState(ServiceButtonState.DISABLED));
+            dispatch(setCargoDoorButtonState(ServiceButtonState.DISABLED));
+            dispatch(setBaggageButtonState(ServiceButtonState.DISABLED));
+            dispatch(setAftDoorButtonState(ServiceButtonState.DISABLED));
+            dispatch(setCateringButtonState(ServiceButtonState.DISABLED));
+            if (cabinDoorOpen === 1) {
+                SimVar.SetSimVarValue('K:TOGGLE_AIRCRAFT_EXIT', 'enum', 1);
+            }
+            if (aftDoorOpen === 1) {
+                SimVar.SetSimVarValue('K:TOGGLE_AIRCRAFT_EXIT', 'enum', 4);
+            }
+            if (cargoDoorOpen === 1) {
+                SimVar.SetSimVarValue('K:TOGGLE_AIRCRAFT_EXIT', 'enum', 6);
+            }
+        } else if (cateringButtonState === ServiceButtonState.DISABLED
```

Comment:
To reduce a lot of this repetition and if this is something you prefer, you can do something like this: 

```typescript
[cateringButtonState, jetWayButtonState, fuelTruckButtonState, gpuButtonState, cargoDoorButtonState, baggageButtonState, aftDoorButtonState, cateringButtonState].every((buttonState) => buttonState === ServiceButtonState.DISABLED);
```